c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1276,17 +1283,20 @@ class DeviceCachingAllocator {
               alloc_block(params, false, context, lock))
           // Free all non-split cached blocks and retry alloc.
           || (C10_LIKELY(captures_underway.empty()) &&
-              release_cached_blocks(context) &&
+              release_cached_blocks(context, {0, 0}) &&
```

Comment:
this is a different behavior introduced by this PR - in about to oom situation, after allocation failed, we release all that we can to release, not just the blocks in the current pool, to increase chances that retrial succeeds.